cmd/tests/cmd_run_test.go

```diff
@@ -1644,3 +1644,75 @@ func TestPrometheusRemoteWriteOutput(t *testing.T) {
 
 	assert.Contains(t, stdout, "output: Prometheus remote write")
 }
+
+func TestBrowserPermissions(t *testing.T) {
+	t.Parallel()
+
+	tests := []struct {
+		name             string
+		envVarValue      string
+		envVarMsgValue   string
+		expectedExitCode exitcodes.ExitCode
+		expectedError    string
+	}{
+		{
+			name:             "no env var set",
+			envVarValue:      "",
+			expectedExitCode: 107,
+			expectedError:    "To run browser tests set env var K6_BROWSER_ENABLE_RUN=true",
+		},
+		{
+			name:             "env var set but set to false",
+			envVarValue:      "false",
+			expectedExitCode: 107,
+			expectedError:    "To run browser tests set env var K6_BROWSER_ENABLE_RUN=true",
+		},
+		{
+			name:             "env var set but set to 09adsu",
+			envVarValue:      "09adsu",
+			expectedExitCode: 107,
+			expectedError:    "To run browser tests set env var K6_BROWSER_ENABLE_RUN=true",
+		},
+		{
+			name:             "with custom message",
+			envVarValue:      "09adsu",
+			envVarMsgValue:   "Try again later",
+			expectedExitCode: 107,
+			expectedError:    "Try again later",
+		},
+		{
+			name:             "env var set and set to true",
+			envVarValue:      "true",
+			expectedExitCode: 0,
+			expectedError:    "",
+		},
+	}
+
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			script := `
+			import { chromium } from 'k6/experimental/browser';
+
+			export default function() {};
+			`
+
+			ts := getSingleFileTestState(t, script, []string{}, tt.expectedExitCode)
+			if tt.envVarValue != "" {
+				t.Setenv("K6_BROWSER_ENABLE_RUN", tt.envVarValue)
```

Comment:
@na-- @mstoykov do you know of a better way to work with env vars that are also exposed to the `RootModule`?

Reply:
I added https://github.com/grafana/k6/pull/2890 to address this problem

Reply:
https://github.com/grafana/k6/pull/2890 was just merged, now you should be able to get the environment from `VU`'s `InitEnv`: 
https://github.com/grafana/k6/blob/134ca65a64f758c19f40fd67b0041b23733ef413/js/modules/modules.go#L44-L45
https://github.com/grafana/k6/blob/134ca65a64f758c19f40fd67b0041b23733ef413/js/common/initenv.go#L19

Reply:
Refactored in 919450d3e632110dd5bff9b3b8d9a508d4f52a11